app/src/main/java/com/amos/flyinn/MainActivity.java

```diff
@@ -175,17 +178,41 @@ public void onClick(View view) {
         this.initViewsWebRTC();
         this.initScreenCapturePermissions();
     }
-
+ 
+ 
+ 
     /**
      * A native method that is implemented by the 'native-lib' native library,
      * which is packaged with this application.
      */
     public native String stringFromJNI();
-
+ 
     public void toScreenActivityOnClick(View view) {
         Intent intent = new Intent(this, RecordingActivity.class);
         startActivity(intent);
     }
+ 
+  
+ 
+    @Override
+    public void onResume()
+    {
+        super.onResume();
+        checkForDebuggingMode();
+    }
+ 
+     private void checkForDebuggingMode() {
+        if(Settings.Secure.getInt(this.getContentResolver(), Settings.Global.ADB_ENABLED, 0) != 1) {
+            try {
+                startActivity(new Intent(android.provider.Settings.ACTION_APPLICATION_DEVELOPMENT_SETTINGS));
+                Toast.makeText(this, "To use FlyInn please enable USB debugging.", Toast.LENGTH_SHORT).show();
+            } catch(Exception ex) {
+                Toast.makeText(this, "To use FlyInn please enable first the debugging mode and then the USB debugging.\n" +
```

Comment:
This is too much text for a toast, especially a short one. An [AlertDialog](https://developer.android.com/guide/topics/ui/dialogs) would be better.

Reply:
I think toasts are okay here, because the user doesn't have a choice what to do. Besides it's faster than to press a button and it forces that flyinn will not be open without the correct settings. 
Regarding unit tests I dont think that it is possible to test the two core functions (get setting configuration and opening settings).

Reply:
This is not about having a choice or not. This is about informing the user what he needs to do in order to get the app running.

Toasts are for short notifications that the user can grasp at first glance and does not necessarily need to read in order to use the app.
But this is much more than just a short notification, this is an instruction which the user not only needs to read but also needs to remember (just for a short time, but the user must remember which setting he needs to set to which value) and that is a process which every user must be able to take as much time for as he needs.
But at the same time the settings activity is opened which is a big and unexpected change in the UI. The main focus of the user will be drawn to the changing app, not some inconspicuous text at the bottom of the display which automatically disappears again before the user gets a real chance of reading it.